tests/integration/sync/test_sync_command.py

```diff
@@ -354,43 +330,50 @@ def _get_api_message(self, rest_api):
                     resource_id = item.get("id")
                     break
             self.api_client.flush_stage_cache(restApiId=rest_api, stageName="prod")
-
-            # RestApi deployment needs a wait time before being responsive
-            time.sleep(API_SLEEP)
-            api_response = self.api_client.test_invoke_method(
-                restApiId=rest_api, resourceId=resource_id, httpMethod="GET"
-            )
-            return api_response.get("body")
         except ClientError as ce:
-            # This test is very unstable, any fixed wait time cannot guarantee a successful invocation
-            if "Invalid Method identifier specified" in ce.response.get("Error", {}).get("Message", ""):
-                LOG.error("The deployed changes are not callable on the client yet, skipping the RestApi invocation")
-                return '{"message": "hello!!"}'
             LOG.error(ce)
         except Exception as e:
             LOG.error(e)
-        return ""
 
-    # def _get_apiv2_message(self, http_api):
-    #     try:
-    #         region = os.getenv("AWS_DEFAULT_REGION", "us-east-1")
-    #         url = f"https://{http_api}.execute-api.{region}.amazonaws.com/beta/hello"
-    #         response = requests.get(url)
-    #         LOG.info(response.text)
-    #     except Exception as e:
-    #         LOG.error(e)
-    #     return ""
+        # RestApi deployment needs a wait time before being responsive
+        count = 0
+        while count < 20:
+            try:
+                time.sleep(1)
+                api_response = self.api_client.test_invoke_method(
+                    restApiId=rest_api, resourceId=resource_id, httpMethod="GET"
+                )
+                return api_response.get("body")
+            except ClientError as ce:
+                # This test is very unstable, any fixed wait time cannot guarantee a successful invocation
+                if "Invalid Method identifier specified" in ce.response.get("Error", {}).get("Message", ""):
```

Comment:
nit: we can LOG the error message here so that it will be visible to person who is investigating possible failure of the test case?